pandas/core/algorithms.py

```diff
@@ -740,8 +739,11 @@ def factorize(
     uniques = _reconstruct_data(uniques, dtype, original)
 
     # return original tenor
-    if isinstance(original, ABCIndexClass):
+    if isinstance(original, ABCIndex):
         if original.dtype.kind in ["m", "M"] and isinstance(uniques, np.ndarray):
+            original._data = cast(
+                "Union[DatetimeArray, TimedeltaArray, np.ndarray]", original._data
+            )
```

Comment:
i think we can rule out ndarray here

Reply:
Thanks, I took another look but I don't see why that's the case. Why is it `_data` can't be an ndarray?

Reply:
since this is a datetimelike subclass by-definition _data is always a datetimelike